### DISCUSSION TITLE
grpc RpcError: [batch_insert], <_MultiThreadedRendezvous: StatusCode.RESOURCE_EXHAUSTED, grpc: received message larger than max (514983585 vs. 67108864)>

### DISCUSSION THREAD

**Discussion by Molly-Xu00** [milvus-io/milvus#30566] [Q&A and General discussion] [answered]
I tried to insert 1million vectors (dimension=128), it reported such error, how to fix it?

**Comment by yhmo** [comment, marked as answer]
By default, the size of RPC request sent to milvus server is limited to 64MB. 
1 million 128 dim vectors, the total size is 1M * 128 * 4bytes = 512MB, it exceeds 64MB.
Split the 1M vectors to batches. Insert 50000 vectors each time.

**Reply by xiaofan-luan** [reply]
for 1M vectors, you could use bulkinsert
